File: YuGiOh2/Cards/C53582587.cs
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 激流葬
    /// </summary>
    public class C53582587
    {
        public static int Type { get; } = (int)AffectMomentType.WhenSummoned;

        public static void ProcessEffect(string targetID, Player player)
        {
            Destroy(player);
            Destroy(player.Enemy);
        }

        private static void Destroy(Player player)
        {
            for (int i = 0; i < 5; i++)
            {
                if (player.Field.MonsterFields[i] == null)
                    continue;

                player.AddCardToGrave(player.Field.MonsterFields[i]);
                player.Field.MonsterFields[i] = null;
            }
        }
    }
}

File: YuGiOh2/Scripts/C53582587.lua
Type = 1;
--激流葬
function processEffect(targetID, player)
    destroy(player.Enemy);
    destroy(player);
end

function destroy(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil then
            player:AddCardToGrave(monster);
            player:ClearMonsterField(i - 1);
        end
    end
end